src/helpers/vertex.ts

```diff
@@ -36,25 +36,31 @@ export class VertexHelper {
     return Object.assign({ payload: JSON.stringify(params) }, baseParams);
   }
 
-  fetchJson(url: string, params: Record<string, unknown>) {
-    return JSON.parse(UrlFetchApp.fetch(url, params).getContentText());
+  fetchJson(url: string, params: Record<string, unknown>): any {
+    const response = UrlFetchApp.fetch(url, params);
+
+    if (response.getResponseCode() === 429) {
+      Utilities.sleep(CONFIG.vertexAi.quotaLimitDelay);
+      return this.fetchJson(url, params);
+    }
+    return JSON.parse(response.getContentText());
   }
 
   predict(prompt: string) {
-    Utilities.sleep(1000); // respect rate limitations
     console.log(`Prompt: ${prompt}`);
 
-    const predictEndpoint = `https://${CONFIG.vertexAi.endpoint}/v1/projects/${this.projectId}/locations/us-central1/publishers/google/models/${CONFIG.vertexAi.modelId}:predict`;
+    const predictEndpoint = `https://us-central1-${CONFIG.vertexAi.endpoint}/v1/projects/${this.projectId}/locations/us-central1/publishers/google/models/${CONFIG.vertexAi.modelId}:predict`;
```

Comment:
should the region be non-configurable?

Reply:
It is, has already been done by Kevin in another change, hence why it's not in mind to avoid merge issues.